MetaMask/metamask-extension
# Integration of the Rave Names Fantom Name System into Address prompts.

[redone]

Explanation:
Integration of the Rave Names FNS into MetaMask's address prompts. Using ethers along with the `rpc.ftm.tools` RPC router to resolve owned names. This comes at no detriment to ENS resolution, or any other part of MetaMask, or development.

Rave Names is a 100% decentralized Web3 username system on Fantom. Any enquiries can be made by replying to this PR or by sending an email to `dev@fantoms.art`, or by sending a direct message to us on twitter @rave_names or https://github.com/FantomsOpera.

Manual testing steps:

- `yarn dist`
- `Click 'Send' at the homescreen to take you to the address prompt screen`
- `Type in z.ftm, which should resolve to 0x3e522051A9B1958Aa1e828AC24Afba4a551DF37d, the public deployer address.`
- `Then, type in randomz.eth which should resolve to the same address.`

Again, feel free to contact us through email or twitter.
